Modules/vc-module-catalog/VirtoCommerce.CatalogModule.Data/ExportImport2/CatalogFullExportPagedDataSource.cs

```diff
@@ -0,0 +1,211 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using VirtoCommerce.CatalogModule.Core.Model;
+using VirtoCommerce.CatalogModule.Core.Model.Search;
+using VirtoCommerce.CatalogModule.Core.Search;
+using VirtoCommerce.CatalogModule.Core.Services;
+using VirtoCommerce.ExportModule.Core.Model;
+using VirtoCommerce.Platform.Core.Assets;
+using VirtoCommerce.Platform.Core.Common;
+
+namespace VirtoCommerce.CatalogModule.Data.ExportImport
+{
+    public class CatalogFullExportPagedDataSource : IPagedDataSource
+    {
+        private class FetchResult
+        {
+            public int TotalCount;
+            public IEnumerable<IExportable> Result = Array.Empty<IExportable>();
+        }
+
+        private class ExportDataSourceState
+        {
+            public FetchResult FetchResult;
+            public SearchCriteriaBase SearchCriteria;
+            public Func<ExportDataSourceState, Task<FetchResult>> FetchFunc;
+
+            public ExportDataSourceState(SearchCriteriaBase searchCriteria, Func<ExportDataSourceState, Task<FetchResult>> fetchFunc)
+            {
+                SearchCriteria = searchCriteria;
+                FetchFunc = fetchFunc;
+            }
+
+            public void SetFetchResult<T>(GenericSearchResult<T> searchResult) where T : Entity
```

Comment:
Same for this - maybe its redundant. For inner class I even do not see necessity in having constructor for properties filling.

Reply:
Removed